perl: Add recipe to build perl

jakirkham opened the pull request:
Builds perl on Mac, Linux, and Windows. Taken from conda recipes. ( https://github.com/conda/conda-recipes/tree/master/perl )

jakirkham commented:
@asmeurer @dan-blanchard, would you have any interest in being added as maintainers of this recipe?

pelson commented:
As a heads up, @msarahan added a perl recipe in https://github.com/conda-forge/staged-recipes/pull/101. It would be good to reconcile the two PRs.

jakirkham commented:
Thanks for the notice. Will combine his changes in the recipe here. Then we can figure out where to go from there.

jakirkham commented:
So I have incorporated @msarahan's changes for Windows here. Also, went ahead and simplified the versioning so it shouldn't be as difficult to maintain.

jakirkham commented:
Have added a PR ( https://github.com/msarahan/staged-recipes/pull/1 ) against @msarahan's PR ( https://github.com/conda-forge/staged-recipes/pull/101 ) with merge conflicts resolved. Can sort out the rest from there. Will close once that is resolved.

pelson commented:
> Will close once that is resolved unless we decide it is simpler to break Perl out and get it in first.

I'm keen to merge once things look good, so for me, a single PR for perl sounds like the preferable option. :+1: on this PR once the CI says :+1: .

jakirkham commented:
Ok, that sounds good. Building perl takes awhile anyways.

Any idea why AppVeyor says it is canceled? Are you restarting it?

pelson commented:
> Any idea why AppVeyor says it is canceled? Are you restarting it?

Sorry about that. I cancelled an earlier commit, but apparently that means that all subsequent commits are automatically cancelled. I've restarted.

pelson commented:
P.S. There are a number of PRs which have put a pretty hefty queue on AppVeyor. It's not ideal, but it should come through in a couple of hours. I'm :+1: on this being merged once that happens.

jakirkham commented:
Oh, ok. Probably for the best as AppVeyor tends to get pretty backlogged.

jakirkham commented:
Sounds good.

BTW, how many AppVeyor build can conda-forge do at a time? Is it just 1 or something larger?

pelson commented:
> BTW, how many AppVeyor build can conda-forge do at a time? Is it just 1 or something larger?

```staged-recipes``` is just 1 I believe. It is my understanding that the feedstocks could all run in parallel theoretically.

msarahan commented:
+1 for splitting it up.  +1 also for building Perl from source.  When I wrote the bit that just repackages strawberry perl, we didn't have great tools for having bash available explicitly (many people installed cygwin or msys, but that was outside of the recipe, and outside of conda.)

I'd be in favor of keeping the build-from-source recipe from conda-recipes if we can - and add other windows dependencies that make the unix-style tools work.  @mingwandroid is working on getting MSYS2 working in Conda.  That might be best, but there might also be a quicker intermediary.

jakirkham commented:
Ok, good, glad you had a chance to look at this, @msarahan .

As the consensus, seems to be to split out Perl, I'll just keep working on this here.

Yeah, I think this was initially expecting msys, but am not sure though. Using Strawberry Perl looks like a good solution and I am happy to make use of it for now. When MSYS2 comes around, we can include a new Windows build recipe, up the Perl version number or build number, and release. If all goes well, then no one will really notice the transition.

FWIW we can always add a PR to the Perl feedstock when it gets generated with the changes for a proper Windows build so they don't get lost.

jakirkham commented:
Passes all tests. :)

jakirkham commented:
@msarahan, are you ok with this as it is now? Certainly will revisit making a better Windows solution once a clear path exists. For now, it looks like your original proposal does quite nicely.

ocefpaf commented on recipes/perl/meta.yaml:
```diff
@@ -0,0 +1,35 @@
+  md5: 7b62cdfc8087b2a0b6a613a4962eebb4  #[win]
+
+build:
+  number: 1
```
`0` :stuck_out_tongue_winking_eye:

jakirkham replied:
Oops. Sorry, I guess it is in now. :/

pelson replied:
Ah. Sorry @ocefpaf. No harm done having a non zero build number though :smile:

ocefpaf replied:
No problem.  I was just messing with @jakirkham :wink:

jakirkham replied:
Will fix this when we next bump the version number.

pelson commented:
Excellent! Thanks @jakirkham

ocefpaf commented on recipes/perl/meta.yaml:
```diff
@@ -0,0 +1,35 @@
+
+source:
+  fn: perl-{{ base_version }}.tar.gz  #[unix]
+  fn:  strawberry-perl-{{ version }}-32bit-portable.zip  #[win]
```
Just curious. What is this?

jakirkham replied:
It was a version of Perl that @msarahan found that is built for Windows. Will hold us over until we can get a proper build system for Windows within conda.

ocefpaf replied:
Cool.

jakirkham commented:
Thanks @pelson and @msarahan.
